### PR TITLE
Make movlhps/movhlps register source operands the correct size.

### PR DESCRIPTION
**Pull request by khuey**
This is necessary to allow encoding of the reg-reg forms of these opcodes.

### CODE REVIEW

**Comment by khuey** [comment]
Hrm, tests seem to pass locally ...

File: suite/tests/api/ir_x86_2args.h [hunk 1]
```diff
@@ -123,8 +123,10 @@ OPCODE(movsd_rr, movsd, movsd, 0, REGARG_PARTIAL(XMM1, OPSZ_8),
        REGARG_PARTIAL(XMM0, OPSZ_8))
 
 OPCODE(movlps, movlps, movlps, 0, REGARG_PARTIAL(XMM0, OPSZ_8), MEMARG(OPSZ_8))
+OPCODE(movlhps, movlps, movlps, 0, REGARG_PARTIAL(XMM0, OPSZ_8), REGARG_PARTIAL(XMM15, OPSZ_8))
```
Comment (derekbruening) [inline suite/tests/api/ir_x86_2args.h]: I'm guessing that the 32-bit tests are failing b/c xmm15 is an invalid register there (not sure why a better error message about failing to encode didn't show up in the test output)

Comment (khuey) [reply suite/tests/api/ir_x86_2args.h]: Ah, that would make sense.  I wanted to test the REX thing but it's not important.

Comment (derekbruening) [reply suite/tests/api/ir_x86_2args.h]: Maybe you could condition it with IF_X64_ELSE(XMM15, XMM7) or sthg

Comment (derekbruening) [reply suite/tests/api/ir_x86_2args.h]: Or use DR_REG_STOP_XMM but have to figure out raw vs the REGARG macros

**Pull request merged**